layers/meta-balena-genericx86/recipes-kernel/linux/linux-yocto_%.bbappend

```diff
@@ -530,3 +546,24 @@ SRC_URI_append_surface-go = " \
     file://0107-media-i2c-Add-support-for-ov5693-sensor.patch \
     file://0108-media-i2c-ov5693-add-delays-to-powerdown.patch \
 "
+
+do_configure_append () {
+    # TODO use the key from balena-keys
+    mkdir -p certs
+    if [ "${SIGN}" != "true" ]; then
+        return 0
+    fi
+
+    export CURL_CA_BUNDLE="${STAGING_DIR_NATIVE}/etc/ssl/certs/ca-certificates.crt"
+
+    RESPONSE_FILE=$(mktemp)
+    curl --fail "${SIGN_API}/kmod/cert/${SIGN_KMOD_KEY_ID}" > "${RESPONSE_FILE}"
+    jq -r .cert "${RESPONSE_FILE}" > certs/balenaos.crt
+    rm -f "${RESPONSE_FILE}"
+}
+
+do_deploy_append() {
+    if [ "${SIGN}" = "true" ]; then
+        install -m 0644 ${B}/${KERNEL_OUTPUT_DIR}/${KERNEL_IMAGETYPE}.initramfs.sig ${DEPLOYDIR}/${KERNEL_IMAGETYPE}.sig
+    fi
+}
```

Comment:
Actually, this do_deploy_append() also needs to be removed as this is done by the sign class already.

Reply:
This is done in https://github.com/balena-os/meta-balena/pull/2399